core/src/tag_utils.rs

```diff
@@ -101,27 +118,44 @@ impl SwfMovie {
         let compressed_len = swf_data.len();
         let swf_buf = swf::read::decompress_swf(swf_data)?;
         let encoding = swf::SwfStr::encoding_for_version(swf_buf.header.version());
-        Ok(Self {
+        let mut movie = Self {
             header: swf_buf.header,
             data: swf_buf.data,
             url,
             loader_url,
             parameters: Vec::new(),
             encoding,
             compressed_len,
-        })
+        };
+        movie.append_parameters_from_url();
+        Ok(movie)
     }
 
     /// Construct a movie based on a loaded image (JPEG, GIF or PNG).
     pub fn from_loaded_image(url: String, length: usize) -> Self {
-        Self {
+        let mut movie = Self {
             header: HeaderExt::default_with_uncompressed_len(length as u32),
             data: vec![],
             url,
             loader_url: None,
             parameters: Vec::new(),
             encoding: swf::UTF_8,
             compressed_len: length,
+        };
+        movie.append_parameters_from_url();
+        movie
+    }
+
+    fn append_parameters_from_url(&mut self) {
+        match Url::parse(&self.url) {
+            Ok(url) => {
+                for (key, value) in url.query_pairs() {
+                    self.parameters.push((key.into_owned(), value.into_owned()));
+                }
+            }
+            Err(e) => {
+                tracing::error!("Failed to parse LoaderInfo parameters from URL: {}", e);
```

Comment:
This error message might make more sense as "Failed to parse self URL" or similar, since the error is to do with the url parsing, not the parameter parsing.